lib/breakfast/exceptions.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Breakfast.CompileError do
+  @moduledoc """
+  Raised at compilation time when the cereal cannot be compiled.
+  """
+  defexception [:message]
+end
+
+defmodule Breakfast.TypeError do
```

Comment:
This _should_ be a `CompileError` instead but since you could call `Breakfast.Type.derive_from_spec/1` at runtime too I thought it was best to have a separate type for it.

Reply:
++